### PR TITLE
fix(theme): fix serving and exporting local themes

### LINKED ISSUE #492: v3.0.3+: Can't serve resume from local theme
## How to reproduce

```
# prepare
git clone https://github.com/paskal/resume.git
docker-compose build build
# run
docker-compose up serve
```

The expected result is resume.json served with jsonresume-theme-kendall-ksenia, the real one is following message in the log plus the try to fetch the theme from the server:

```
TypeError: renderHtml is not a function
    at /data/node_modules/resume-cli/build/builder.js:50:26
    at FSReqCallback.readFileAfterClose [as oncomplete] (internal/fs/read_file_context.js:63:3)
Could not run the render function from local theme.
(node:1) UnhandledPromiseRejectionWarning: TypeError: Cannot read property '1' of null
    at denormalizeTheme (/data/node_modules/resume-cli/build/builder.js:14:46)
    at sendExportHTML (/data/node_modules/resume-cli/build/builder.js:20:30)
    at /data/node_modules/resume-cli/build/builder.js:58:7
    at FSReqCallback.readFileAfterClose [as oncomplete] (internal/fs/read_file_context.js:63:3)
(node:1) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:1) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
<...>
Requesting theme from server...
```

I assume `renderHtml is not a function` is not something I am supposed to see while running the `serve` command, might be a bug.

Serve settings:
```yaml
version: '2'
services:

    serve:
        image: paskal/jsonresume
        working_dir: /data/node_modules/jsonresume-theme-kendall-ksenia
        command: serve --theme .
        volumes:
            - ./resume.json:/data/node_modules/jsonresume-theme-kendall-ksenia/resume.json:ro
        ports:
            - "4000:4000"
```

### LINKED ISSUE #538: Local themes broken again
Shortly after 2ec202b25c7c1fcd825e9fb1a4b9f36cf9157a07 fixed #386, #446 broke them again.

If you call `resume export some.pdf -t ./path/to/theme -r resume.json` it raises the error
```
TypeError: Cannot read property 'join' of undefined
```

I've created #537 as a potential fix.

### PR DESCRIPTION
**Pull request by XuluWarrior**
Fixes #492, fixes #538.

### CODE REVIEW

File: lib/render-html.test.js
Comment (XuluWarrior) [inline lib/render-html.test.js:7]: While writing the new tests, I found that this mock of resolve is not called.
So tests are running against the actual file system.
Possibly the mock can be removed.  Or maybe it needs to be fixed.

File: lib/builder.js
Comment (XuluWarrior) [inline lib/builder.js:6]: This project uses a mix of CommonJS and ES6 modules.  This should probably be made consistent to prevent bugs like this.
I won't address it in this PR.

**Comment by wxsms** [comment]
This solution works. thank you.

File: lib/render-html.test.js
Comment (XuluWarrior) [reply lib/render-html.test.js:44]: The test below is a valid path.
It would fail pre my changes.

**Comment by davidNHK** [comment]
Can i kwon will it merge soon ?

**Comment by michaltalaga** [comment]
bump

**Comment by XuluWarrior** [comment]
Are there any project maintainers about to review/comment on this PR?  I see that there haven't been any merges for a few months.

File: lib/render-html.test.js
Comment (XuluWarrior) [reply lib/render-html.test.js:7]: The PR comment is an implicit question about what the correct action is.
Any TODO would come from the answer to the question.
And I'm not sure that in code TODOs is the way this project manages future code changes.  They may prefer creating a GitHub issue.

**Comment by VCNinc** [comment]
Is this project abandoned?

**Comment by VCNinc** [comment]
@thomasdavis

**Comment by alexkreidler** [comment]
I'm getting `TypeError: renderHtml is not a function`, it seems like this PR would fix that.

It would be great to get this merged. I'll ping the top commiters, so those who have commit rights and some free time can take a look: @rolandnsharp @thomasdavis @PeterDaveHello @rbardini @erming @antialias 

Also if someone has time to run `upgrade-interactive` to bump the dependencies then you can close all the dependabot PRs.

Thanks all for this project!

**Comment by paskal** [comment]
Gentle ping  @rolandnsharp @thomasdavis @PeterDaveHello @rbardini @erming @antialias

**Review by paskal — APPROVED** [review]

**Comment by nhymxu** [comment]
I don't have problem with local theme.
Just problem with `const renderHtml = require('./render-html').default;`

So, for clear and small PR. Should we split to two PR?

**Comment by XuluWarrior** [comment]
So now @paskal has approved this PR what is the next step?
Does it require further approval?
Or somebody with merge rights to do the merge?

**Comment by paskal** [comment]
I am no one here. Someone from the list above needs to approve and merge it.

**Review by rbardini — APPROVED** [review]

**Comment by rbardini** [comment]
Thanks @XuluWarrior for fixing this, and sorry everyone for the delay! As a side note, I've been writing an [alternative JSON Resume builder](https://github.com/rbardini/resumed) from scratch that I hope might shape the future of resume-cli, in case anyone wants to give it a try 🙂

**Pull request merged**
